### PR TITLE
bugfix for empty_

### PR DESCRIPTION
**Pull request by youkaichao**
- [ ] I have marked all applicable categories:
    + [ ] exception-raising fix
    + [ ] algorithm implementation fix
    + [ ] documentation modification
    + [ ] new feature
- [ ] If applicable, I have mentioned the relevant/related issue(s)

Less important but also useful:

- [ ] I have visited the [source website], and in particular read the [known issues]
- [ ] I have searched through the [issue tracker] and [issue categories] for duplicates
- [ ] I have mentioned version numbers, operating system and environment, where applicable:
  ```python
  import tianshou, torch, sys
  print(tianshou.__version__, torch.__version__, sys.version, sys.platform)
  ```

  [source website]: https://github.com/thu-ml/tianshou
  [known issues]: https://github.com/thu-ml/tianshou/#faq-and-known-issues
  [issue categories]: https://github.com/thu-ml/tianshou/projects/2
  [issue tracker]: https://github.com/thu-ml/tianshou/issues?q=

### CODE REVIEW

**Comment by youkaichao** [comment]
``np.isnan`` does not support string argument, so the if-branch only works for numbers.
For bool type, ``*= 0`` would change the data type, and ``&=False`` suits better.

File: tianshou/data/batch.py [hunk 1]
```diff
@@ -569,10 +569,12 @@ def empty_(self, index: Union[
             else:  # scalar value
                 warnings.warn('You are calling Batch.empty on a NumPy scalar, '
                               'which may cause undefined behaviors.')
-                if isinstance(v, (np.generic, Number)):
+                if isinstance(v, (np.number, Number)):
```
Comment (duburcqa) [inline tianshou/data/batch.py]: Why not to add `np.bool_` here. `self.__dict__[k] *= 0` is doing the same as `self.__dict__[k] &= False`, so I don't see the point in creating a new if case.

Comment (duburcqa) [reply tianshou/data/batch.py]: OK I just saw your comment. There is the same problem with native Python `bool` type. So maybe you were right in the firstt place, and using `self.__dict__[k] = v.__class__(0)` is the most robust approach after all.

Comment (youkaichao) [reply tianshou/data/batch.py]: Agree. Although ``self.__dict__[k] = v.__class__(0)`` will lead to new memory allocation, the memory cost of scalar is negligible.

**Review by duburcqa — APPROVED** [review]

File: tianshou/data/batch.py (hunk 1 above)
Comment (duburcqa) [reply tianshou/data/batch.py]: Indeed.

**Review by Trinkle23897 — APPROVED** [review]

**Pull request merged**